client/me/two-step/index.jsx

```diff
@@ -39,6 +40,24 @@ class TwoStep extends Component {
 		this.props.fetchUserSettings();
 	};
 
+	componentDidMount() {
+		if ( ! this.props.isFetchingUserSettings && ! this.props.isEmailVerified ) {
+			window.location.href = '/me/security';
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		// Check if fetching just completed
+		if (
+			prevProps.isFetchingUserSettings &&
+			! this.props.isFetchingUserSettings &&
+			prevProps.isEmailVerified &&
+			! this.props.isEmailVerified
+		) {
+			window.location.href = '/me/security';
```

Comment:
Using `page( '/me/security' )` would be much faster.

Reply:
Opted for `page.redirect` since it doesn't include the path in the history.